src/tools/Circle.ts

```diff
@@ -24,13 +24,22 @@ export default class Circle implements CircleInterface {
       this.showRadius)
   }
 
-  // eslint-disable-next-line
-  mouseDownAction = (event: CustomEvent, canvasElement: CanvasElement, layer: Konva.Layer, socket: WebSocket): void => {
-    canvasElement.data = [event.globalOffset.x, event.globalOffset.y]
-    canvasElement.id = uuid()
-    canvasElement.hasMoved = false
-    canvasElement.tracker = Tracker.ADDITION
-    canvasElement.tool = {
+  mouseDownAction = (event: CustomEvent): void => {
+    this.enableTool()
+    this.resetCanvasEntity()
```

Comment:
Here we give our temporary data to the CanvasEntity module. We first reset it so that we dont get the data from an earlier use.